orttraining/orttraining/python/orttraining_pybind_state.cc

```diff
@@ -75,6 +76,30 @@ void ResolveExtraProviderOptions(const std::vector<std::string>& provider_types,
   }
 }
 
+std::vector<std::shared_ptr<IExecutionProvider>> GetExecutionProviders(int device_type, int device_id) {
```

Comment:
May we call it more explicitly as `GetExecutionProvidersForTrainingApis(...)`? And let's also put some documentation at the beginning to tell readers what the function is doing.

Reply:
Is this only for on-device training? should we put it under ENABLE_TRAINING_ON_DEVICE?